jetty-server/src/main/java/org/eclipse/jetty/server/SymlinkAllowedResourceAliasChecker.java

```diff
@@ -45,6 +45,9 @@ public SymlinkAllowedResourceAliasChecker(ContextHandler contextHandler)
     @Override
     protected boolean check(String pathInContext, Path path)
     {
+        if (_base == null)
+            return false;
```

Comment:
Ditto here, if a user adds this AliasChecker, but it's not being used due to an invalid base, it should indicate as such in the log.warn.

Warning once during initialization should be sufficient to convey this state.

Reply:
I don't think we should warn here. This alias checker is always added by default on a `ContextHandler` if `File.separatorChar == '/'`, I think there are valid uses of contextHandler that don't need a base resource so we don't want it always printing out warnings.

Reply:
I'm not saying to warn here.
Warn during initialization (aka doStart())

> I think there are valid uses of contextHandler that don't need a base resource so we don't want it always printing out warnings.

This is a bad nuance, and I feel it is a bad argument too.

There's only 2 things in `ContextHandler` that uses the Base Resource.  

1. the `toString()` implementation.
2. the `getResource()` implementation.

IMO ...
The code path that adds the symlink alias checker should actually be in doStart(), not in the constructor of ContextHandler.
And it shouldn't add the alias checker unless there is a base resource.
And it shouldn't allow changing the base resource after start.

Reply:
> This is a bad nuance, and I feel it is a bad argument too.

We have implementations of `ContextHandler` that aren't supposed to be used with a base resource. For example `MovedContextHandler`, should you be required to set a base resource to not get a warning with this. Same goes for `HttpSpiContextHandler` and the `WellKnownHandler` (still in PR). As far as I'm aware all of these have no use for baseResource.

> The code path that adds the symlink alias checker should actually be in doStart(), not in the constructor of ContextHandler.

The problem is if they don't want the symlink alias check, previously they could just clear it after the constructor. At the point of `doStart()` we don't know if they wanted the symlink check or not. Did they clear the alias checker list or was the list always empty, did they add an addition to the list or did they clear the list and add a replacement alias check.

```java
ServletContextHandler contextHandler = new ServletContextHandler();
contextHandler.clearAliasChecks();
contextHandler.addAliasCheck(customAliasCheck);
```

Maybe we shouldn't change this behaviour in 9.4.

Reply:
This is still possible to support with the move of the check to ContextHandler.doStart()

Reply:
I think no matter what, changing to adding checkers in doStart is going to be too much of a behaviour change for 9.4.  As @lachlan-roberts says, we can't work out if they added a checker that wasn't the one we would add if they actually want it added or not.

So let's do minimal changes for 9.4.

For 10, 11 or 12, I'm OK with some behaviour changes... but I would prefer those to be that we don't add any aliase checkers by default.